src/Vizkit3dWorld.cpp

```diff
@@ -18,27 +18,37 @@
 
 namespace vizkit3d_world {
 
-Vizkit3dWorld::Vizkit3dWorld(std::string path, 
+Vizkit3dWorld::Vizkit3dWorld() {
+
+}
+
+Vizkit3dWorld::~Vizkit3dWorld()
+{
+}
+
+void Vizkit3dWorld::initialize(std::string path,
                             std::vector<std::string> modelPaths,
                             std::vector<std::string> ignoredModels,
-                            int cameraWidth, int cameraHeight, 
-                            double horizontalFov, 
+                            int cameraWidth, int cameraHeight,
+                            double horizontalFov,
                             double zNear, double zFar)
-    : worldPath(path)
-    , widget(NULL)
-    , modelPaths(modelPaths)
-    , app(NULL)
-    , cameraWidth((cameraWidth <= 0) ? 800 : cameraWidth)
-    , cameraHeight((cameraHeight <= 0) ? 600 : cameraHeight)
-    , zNear(zNear)
-    , zFar(zFar)
-    , horizontalFov(horizontalFov)
 {
+
+    worldPath = path;
+    widget = NULL;
+    modelPaths = modelPaths;
+    cameraWidth = (cameraWidth <= 0) ? 800 : cameraWidth;
+    cameraHeight = (cameraHeight <= 0) ? 600 : cameraHeight;
+    zNear = zNear;
+    zFar = zFar;
+    horizontalFov = horizontalFov;
+
     loadGazeboModelPaths(modelPaths);
 
     int argc = 1;
     char const*argv[] = { "vizkit3d_world" };
-    app = new QApplication(argc, const_cast<char**>(argv));
```

Comment:
QApplication requires that argc and argv remain valid for the entire lifetime of the QApplication instance, therefore you cannot declare them locally in this scope. 

Reference: http://doc.qt.io/qt-5/qcoreapplication.html#QCoreApplication